llvm-tools/src/main/scala/scala/scalanative/llvm/LLVM.scala

```diff
@@ -0,0 +1,158 @@
+package scala.scalanative
+package llvm
+
+import java.io.File
+
+import scala.sys.process.{Process, ProcessLogger}
+import scala.util.Try
+
+import System.{lineSeparator => nl}
+
+object LLVM {
```

Comment:
The goal is to extract all the code that is purely to call LLVM from the sbt plugin. This will allow us to reuse it from our tests, for instance. In the end, we should be able to run the complete compilation pipeline and observe the runtime behavior of the compiled program from the tests. For instance, one could compile the same code with 2 different drivers and verify that the second driver leads to less `Load`s executed at runtime.

I don't agree with your comment regarding the separation. For me, `unpackNativelib` happens to use LLVM, but it's not LLVM's responsibility to extract the `nativelib`. What are the other parts that seem out of place?